### PR TITLE
[Backport] Remove documentation hints, ref #1536

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Backport and fixed some diffs between v19/v20

### Related Pull Requests

1. See OpenMage/magento-lts#1536
2. See OpenMage/magento-lts#1610
3. See OpenMage/magento-lts#1873

### Questions or comments

@fballiano merge to v20 pls too.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by fballiano** [comment]
couldn't we just cherry-pick [ced97cd](https://github.com/OpenMage/magento-lts/commit/ced97cd763c68f39aa5a5afcb44889fd8389bbe8)?

**Comment by sreichel** [comment]
No, b/c #1610. Also added/reverted/syncd some other changes betwenn 19/20.

**Review by ADDISON74 — APPROVED** [review]

File: lib/Zend/Registry.php [hunk 1]
```diff
@@ -195,16 +195,4 @@ public function __construct($array = array(), $flags = parent::ARRAY_AS_PROPS)
...
     }
 
-    /**
-     * @param string $index
-     * @returns mixed
-     *
-     * Workaround for http://bugs.php.net/bug.php?id=40442 (ZF-960).
-     */
-    #[ReturnTypeWillChange]
-    public function offsetExists($index)
```
Comment (fballiano) [inline lib/Zend/Registry.php:205]: why this change?

File: js/varien/js.js [hunk 2]
```diff
@@ -609,18 +609,6 @@ Element.addMethods({
...
-{
-    var names = ["log", "debug", "info", "warn", "error", "assert", "dir", "dirxml",
-    "group", "groupEnd", "time", "timeEnd", "count", "trace", "profile", "profileEnd"];
-
-    window.console = {};
-    for (var i = 0; i < names.length; ++i)
-        window.console[names[i]] = function() {}
-}
-*/
-
```
Comment (ADDISON74) [inline js/varien/js.js:623]: This change should be discussed. It is not related to this PR.

Comment (fballiano) [reply js/varien/js.js:623]: it is commented code so it's fine to remove it altough not part of the specific scope of the PR

File: lib/Zend/Registry.php (hunk 1 above)
Comment (sreichel) [reply lib/Zend/Registry.php:205]: See OpenMage/magento-lts#1873

**Review by fballiano — APPROVED** [review]
